command_line/method_parser.py

```diff
@@ -0,0 +1,100 @@
+import inspect
+import re
+from collections import defaultdict
+
+from command_line.parser import Parser, Argument
+
+
+def analyze_docstring(docstring: str):
+    """Analyze docstring and collect arguments with descriptions.
+
+    All arguments have to start with lowercase, be followed with
+    colon (:) and description of the argument.
+    """
+    help_strings = defaultdict(list)
+    collect_help = False
+    definition = re.compile(r'(?P<name>.+?):(?P<value>.*)')
+    args_sections = ['Arguments:', 'Ags:']
+    empty_lines = 0
+    argument = None
+
+    for line in docstring.split('\n'):
+        line = line.strip()
+
+        if not line:
+            empty_lines += 1
+            collect_help = False
+        else:
+            empty_lines = 0
+
+        if any(line.startswith(section) for section in args_sections):
+            collect_help = True
+            continue
+
+        if collect_help:
+            match = definition.match(line)
+            if match:
+                argument = match.group('name')
+                if argument[0].isupper():
+                    collect_help = False
+                    continue
+                value = match.group('value').lstrip()
+                if value:
+                    help_strings[argument].append(value)
+            elif argument:
+                help_strings[argument].append(line)
+
+    for key, value in help_strings.items():
+        help_strings[key] = ' '.join(value)
+
+    return help_strings
+
+
+def is_set(value):
+    return not (value == inspect._empty)
+
+
+def empty_to_none(value):
+    if value == inspect._empty:
```

Comment:
Maybe use the above `is_set` function in `empty_to_none` because it is already written? like: 
`if is_set(value):
    return value
else:
    return None`
just a proposition to consider this